src/neo4j_graphrag/experimental/pipeline/pipeline.py

```diff
@@ -182,40 +189,100 @@ def show_as_dict(self) -> dict[str, Any]:
         return pipeline_config.model_dump()
 
     def draw(
-        self, path: str, layout: str = "dot", hide_unused_outputs: bool = True
+        self, path: str, layout: str = "force", hide_unused_outputs: bool = True
     ) -> Any:
-        G = self.get_pygraphviz_graph(hide_unused_outputs)
-        G.layout(layout)
-        G.draw(path)
+        """Draw the pipeline graph using neo4j-viz.
 
-    def get_pygraphviz_graph(self, hide_unused_outputs: bool = True) -> pgv.AGraph:
-        if pgv is None:
+        Args:
+            path (str): Path to save the visualization. If the path ends with .html, it will save an HTML file.
+                Otherwise, it will save a PNG image.
+            layout (str): Layout algorithm to use. Default is "force".
+            hide_unused_outputs (bool): Whether to hide unused outputs. Default is True.
+
+        Returns:
+            Any: The visualization object.
+        """
+        G = self.get_neo4j_viz_graph(hide_unused_outputs)
+        if path.endswith(".html"):
+            # Save as HTML file
+            with open(path, "w") as f:
+                f.write(G.render()._repr_html_())
+        else:
+            # For other formats, we'll use the render method and save the image
+            G.render()
+            # Note: neo4j-viz doesn't support direct saving to image formats
```

Comment:
Is the comment saying that it can't save to PNG? Because code in the 'else' looks very similar to the one in the former 'if' block to me.